### PR TITLE
Fix #361; Compare GRU's vs Current LSTM

### LINKED ISSUE #361: Compare GRU's vs Current LSTM's

### PR DESCRIPTION
**Pull request by andi4191**
**Note: This branch is not intended for merging into the master branch. It is simply a part of benchmark variation comparison. Hence, the branch name is prefixed with test_**

Test with GRU cell instead of Current LSTM cell

GRU cell results:

Dataset used: TED
Training Time: 8:13:07 (Lower than usual which is expected as compared to LSTMs)
No. of Train epochs: 10
GPU instances used: 4

I FINISHED Optimization - training time: 8:13:07
I Test of Epoch 10 - WER: 0.530035, loss: 94.745821317, mean edit distance: 0.254709
I --------------------------------------------------------------------------------
I WER: 0.200000, loss: 5.857036, mean edit distance: 0.050000
I  - src: "to do that dont look"
I  - res: "o do that dont look"
I --------------------------------------------------------------------------------
I WER: 0.333333, loss: 4.066966, mean edit distance: 0.111111
I  - src: "and i say"
I  - res: "and i saw"
I --------------------------------------------------------------------------------
I WER: 0.333333, loss: 4.303623, mean edit distance: 0.090909
I  - src: "yes he said"
I  - res: "yes the said"
I --------------------------------------------------------------------------------
I WER: 0.333333, loss: 5.691000, mean edit distance: 0.222222
I  - src: "and i say"
I  - res: "and i save"
I --------------------------------------------------------------------------------
I WER: 0.500000, loss: 4.931426, mean edit distance: 0.285714
I  - src: "sort of"
I  - res: "sort a"
I --------------------------------------------------------------------------------
I WER: 0.500000, loss: 5.437888, mean edit distance: 0.133333
I  - src: "not intensively"
I  - res: "not intensely"
I --------------------------------------------------------------------------------
I WER: 1.000000, loss: 1.898744, mean edit distance: 0.333333
I  - src: "and"
I  - res: "nd"
I --------------------------------------------------------------------------------
I WER: 1.000000, loss: 2.118078, mean edit distance: 0.500000
I  - src: "so"
I  - res: "of"
I --------------------------------------------------------------------------------
I WER: 1.000000, loss: 2.586474, mean edit distance: 0.500000
I  - src: "so"
I  - res: "o"
I --------------------------------------------------------------------------------
I WER: 1.000000, loss: 3.953010, mean edit distance: 0.500000
I  - src: "so"
I  - res: "o"
I --------------------------------------------------------------------------------

Will post the results from Current LSTM cell test run after a couple of more benchmark variation algorithm test runs.

### CODE REVIEW

**Comment by kdavis-mozilla** [comment]
When comparing _X_ against _Y_ you run both for 10 epochs.

How do you know that _X_ and _Y_ do not converge at different rates? In other words _X_ may take 1000 epochs to converge. _Y_ may take 100. But _X_ may have a lower WER when it converges. Thus at epoch 10 _X_ may look much worse than _Y_ as it's no where near convergence. However, _X_ may in fact be much better than _Y_ when it converges.

How do you know you have the optimal hyperparameters for both _X_ and _Y_? Did you check them?

**Comment by andi4191** [comment]
@kdavis-mozilla : Yes, I called GRuNN from above two functions in my test run branch but removed it from the PR. 

PS: Also, I am going to move these branches from here to my forked version as these are test branches
